lib/promise/group.rb

```diff
@@ -0,0 +1,39 @@
+class Promise
+  class Group
+    attr_reader :promise
+
+    def initialize(inputs)
+      @remaining = 0
+      @promise = Promise.new
+      @inputs = inputs
+      @on_fulfill = method(:on_fulfill)
+      @on_reject = promise.method(:reject)
+      chain_inputs(inputs)
+    end
+
+    private
+
+    def chain_inputs(inputs)
+      inputs.each do |obj|
+        if promise?(obj)
```

Comment:
`next unless` would drop an indentation level though I'm not sure it actually reads any cleaner, your call